internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -729,6 +730,22 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 	return reconcileComplete{dirty: true, stop: true}
 }
 
+func (r *BMACReconciler) isNonePlatform(ctx context.Context, cd *hivev1.ClusterDeployment) (isNone bool, err error) {
+	if cd.Spec.ClusterInstallRef == nil {
+		return false, errors.Errorf("Cluster Install Reference is null for cluster deployment ns=%s name=%s", cd.Namespace, cd.Name)
```

Comment:
Does this have to be an error? If it's an error, we may end in an infinite reconcile loop. We could also not return an error, call `reconcileComplete` and let a new reconcile happen if there are changes to the Clusterdeployment. [BMAC has a watch on ClusterDeployments](https://github.com/openshift/assisted-service/blob/master/internal/controller/controllers/bmh_agent_controller.go#L1402-L1404)

Reply:
In this case we will have to return something to indicate that we do not want to reconcile again